lib/infomon.rb

```diff
@@ -0,0 +1,515 @@
+# frozen_string_literal: true
+
+# Replacement for the veneable infomon.lic script used in Lich4 and Lich5 (03/01/23)
+# Supports Ruby 3.X builds
+#
+#     maintainer: elanthia-online
+#   contributers: Tillmen, Shaelun, Athias
+#           game: Gemstone
+#           tags: core
+#       required: Lich > 5.6.2
+#        version: 2.0
+#         Source: https://github.com/elanthia-online/scripts
+
+require 'English'
+
+module Infomon
+  $infomon_debug = false
+  setpriority(1)
+
+  class Monitor
+    require "#{LIB_DIR}/stats-info.rb"
+    sleep(0.1) until Char.name && !Char.name.empty?
+    # Need to confirm if this approach disrupts DR start-up
+    if XMLData.game =~ /^(?:GSF|GSIV|GSPlat|GST|Test)$/
+      _respond 'Moving forward ho!' if $infomon_debug
+    else
+      echo "This script is meant for Gemstone Prime, Platinum, or Shattered.  It will likely cause problems on whatever game you're trying to run it on..."
+      exit # probably should unload rather than exit, or DR will never get a connection
+    end
+
+    sleep(0.1) until Char.name && !Char.name.empty?
+
+    # This section will require specific calls to Lich.db rather than relying on CharSettings,
+    # unless we overload the CharSettings section to take a specific script name (infomon)
+
+    @save_value = {}
+    $infomon_values = DB_Store.read('infomon')
+    pp $infomon_values if $infomon_debug
+    # sleep 0.5
+    respond 'starting sequence' if $infomon_debug
+    respond $infomon_values if $infomon_debug
+    echo $infomon_values['active_spells'] if $infomon_debug
+    echo $infomon_values['need_psm_update'] if $infomon_debug
+    echo $infomon_values['Stats'] if $infomon_debug
+
+    respond 'Calling active spells' if $infomon_debug
+    $infomon_values['active_spells'] = {} unless $infomon_values['active_spells'].instance_of?(Hash)
+
+    respond 'Calling need psm update' if $infomon_debug
+    $infomon_values['need_psm_update'] = true if $infomon_values['need_psm_update'].nil?
+
+    #
+    # Load spell info
+    #
+    unless Spell.load
+      echo 'error: failed to load spell list'
+      exit
```

Comment:
We should not do this, this is exit the Process

Reply:
Scoped the library back to being required to be loaded from a script, `infomon.lic` for now. Which allows for safe `exit` on this failure.